Updated regex in StringUtil.parseLinks

jcq opened the pull request:
### Description
This adjusts the regex used in the `parseLinks` function from frontend `StringUtil.js` and adds a couple of Jest tests.

### Background
- [Slack thread](https://dsva.slack.com/archives/CQNPUG8EQ/p1595964938014300)

The previous version was crashing when it encountered a newline char at the end of what should be the parsed URL:
```
motion:\nhttps://appeals.cf.ds.va.gov/reader/appeal/526e114b-af2d-4bc2-90ec-1ab1da0176b1/documents/42416990\nfoo
```

### Acceptance Criteria
- [ ] Code compiles correctly
- [ ] Strings are properly parsed without crashing

jcq commented:
Ugh, @nanotone is right — the regex had a number of problems from the get-go and currently still is failing with query strings; that's my bad for throwing only a limited set of tests at it. :(

Will look to make it more robust

jcq commented:
@nanotone I think I'm going to switch to this one:
https://regex101.com/r/ihvMDa/1

nanotone commented:
@jcq I'm kinda wary of switching to a bigger and more complex regex, since regex has an inverse correlation between length and readability -- I definitely wouldn't want to get a bug report like "user pasted a proper URL but it didn't get converted to a link."

This regex comes with a _lot_ of bells and whistles, including stuff that I doubt we'll ever need:
- ftp:// and unspecified schemes
- username and password
- IP addresses, with logic to exclude local-network IP ranges (!)
- non-ASCII characters in the domain name
- non-standard port number

On the other hand, I think the original regex is easily salvageable by using parts of this one:

```
(https?:\/\/|\b)[-\da-z.]+\.[a-z]{2,6}\/\S*
```

English translation: `https?://` or word-boundary; a domain ending in a 2-6 letter TLD; `/`; any number of non-whitespace chars

jcq commented:
@nanotone I think that's a fair assessment. I grabbed that one as it was rated as the most comprehensive amongst a large set of possibilities (handling the most edge cases), but in our case, we don't really need that much versatility.

I've updated it to use the simpler regex you suggested (and updated a couple other bits).

nanotone commented:
@jcq I think this looks good! Sorry for being a bit pushy on that other regex 😬 

We should aim to get this merged today, but I'd like one other pair of 👀  to go over this for perspective. Tagging Echo 🐬 folks who were in the Slack thread.

lomky commented on client/app/util/StringUtil.js:
```diff
@@ -89,12 +89,25 @@ const StringUtil = {
-    // eslint-disable-next-line no-useless-escape
-    const regex = /(https?:\/\/)?([\da-z\.-]+)\.([a-z\.]{2,6})([\/\w \.-]*)*\/?$/gi;
+    // eslint-disable-next-line
+    const regex = /(https?:\/\/|\b)[-\da-z.]+\.[a-z]{2,6}\/\S*/gi;
```
Can we add a comment for our regex source? always nice to have ~6months later when you're coming back to it.

lomky replied:
oh, I see we're tweaking this one. Maybe just that line Yang added with the English translation?

jcq replied:
> Can we add a comment for our regex source? always nice to have ~6months later when you're coming back to it.

Sure, I can put a note in about the source of the original regex and that it was modified by @nanotone
